CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java

```diff
@@ -145,14 +144,19 @@ protected void componentShowing() {
 
     @Override
     protected void handleNewGraph(final Graph graph) {
+        System.setProperty("dav.graph.ready", Boolean.FALSE.toString());
         if (needsUpdate() && getDataAccessPane() != null) {
             getDataAccessPane().update(graph);
-            Platform.runLater(() -> 
+            Platform.runLater(() ->
                 DataAccessUtilities.loadDataAccessState(getDataAccessPane(), graph)
             );
         }
+        Platform.runLater(() ->
+                // nested runLater so it runs after all the other triggered processes for opening a graph have been run
+                Platform.runLater(() -> System.setProperty("dav.graph.ready", Boolean.TRUE.toString()))
+        );
```

Comment:
I might need some help understanding this one. To my understanding, when you call `Platform.runLater`, it should add the task to the end of the queue so if you are only wanting to wait for the line above to execute, that should already happen without the need to add a nested one

Reply:
Yes, the task will be added to the queue, but too early in the queue. There will be other events fired and added to the queue as well due to the graph being loaded. When it starts processing the queue, we want all those other queued entries to run first, hence the need for the nested `Platform.runLater` call. This effectively puts the call at the end of the "graph loading" queue.